Source/AssetRipper.Processing/AnimatorControllers/AnimatorStateContext.cs

```diff
@@ -0,0 +1,365 @@
+
+		private static IBlendTree CreateBlendTree(ProcessedAssetCollection virtualFile, IAnimatorController controller, IStateConstant state, int nodeIndex)
+		{
+			IBlendTree blendTree = virtualFile.CreateAsset((int)ClassIDType.BlendTree, BlendTree.Create);
+			blendTree.HideFlagsE = HideFlags.HideInHierarchy;
+
+			IBlendTreeNodeConstant node = state.GetBlendTree().NodeArray[nodeIndex].Data;
+
+			blendTree.Name = BlendTreeName;
+
+			blendTree.Childs.Capacity = node.ChildIndices.Count;
+			for (int i = 0; i < node.ChildIndices.Count; i++)
+			{
+				AddAndInitializeNewChild(virtualFile, controller, state, blendTree, nodeIndex, i);
+			}
+
+			if (node.BlendEventID != uint.MaxValue)
+			{
+				blendTree.BlendParameter = controller.TOS[node.BlendEventID];
+			}
+			if (node.BlendEventYID != uint.MaxValue)
+			{
+				blendTree.BlendParameterY = controller.TOS[node.BlendEventYID];
+			}
+			blendTree.MinThreshold = node.GetMinThreshold();
+			blendTree.MaxThreshold = node.GetMaxThreshold();
+			blendTree.UseAutomaticThresholds = false;
+			blendTree.NormalizedBlendValues = node.BlendDirectData?.Data.NormalizedBlendValues ?? false;
+			if (blendTree.Has_BlendType_Int32())
+			{
+				blendTree.BlendType_Int32 = (int)node.BlendType;
+			}
+			else
+			{
+				blendTree.BlendType_UInt32 = node.BlendType;
+			}
+			return blendTree;
+		}
+
+		private static IChildMotion AddAndInitializeNewChild(ProcessedAssetCollection file, IAnimatorController controller, IStateConstant state, IBlendTree tree, int nodeIndex, int childIndex)
+		{
+			IChildMotion childMotion = tree.Childs.AddNew();
+			IBlendTreeConstant treeConstant = state.GetBlendTree();
+			IBlendTreeNodeConstant node = treeConstant.NodeArray[nodeIndex].Data;
+			int childNodeIndex = (int)node.ChildIndices[childIndex];
+			IMotion? motion = CreateMotion(file, controller, state, childNodeIndex);
+			childMotion.Motion.SetAsset(tree.Collection, motion);
+
+			IBlendTreeNodeConstant childNode = treeConstant.NodeArray[childNodeIndex].Data;
+			if (childNode.IsBlendTree())
+			{
+				// BlendTree ChildMotions are not allowed to use TimeScale or Mirror
+				// https://github.com/Unity-Technologies/UnityCsReference/blob/4e215c07ca8e9a32a589043202fd919bdfc0a26d/Editor/Mono/Inspector/BlendTreeInspector.cs#L1469
+				// https://github.com/Unity-Technologies/UnityCsReference/blob/4e215c07ca8e9a32a589043202fd919bdfc0a26d/Editor/Mono/Inspector/BlendTreeInspector.cs#L1488
+				childMotion.TimeScale = 1;
+				childMotion.Mirror = false;
+			}
+			else
+			{
+				childMotion.TimeScale = 1 / childNode.Duration;
+				childMotion.Mirror = childNode.Mirror;
+			}
+			childMotion.CycleOffset = childNode.CycleOffset;
+
+			childMotion.Threshold = node.GetThreshold(childIndex);
+			childMotion.Position?.CopyValues(node.GetPosition(childIndex));
+			if (node.TryGetDirectBlendParameter(childIndex, out uint directID))
+			{
+				childMotion.DirectBlendParameter = controller.TOS[directID];
+			}
+
+			return childMotion;
+		}
```

Comment:
Conceptually, these static creation methods don't seem to be conceptually part of this class. Please move them back to VirtualAnimationFactory. Adjust accessibility if necessary.